rollup/tracing/tracing.go

```diff
@@ -579,24 +582,27 @@ func (env *TraceEnv) fillBlockTrace(block *types.Block) (*types.BlockTrace, erro
 			CodeSize:         statedb.GetCodeSize(env.coinbase),
 		},
 		Header:            block.Header(),
+		Bytecodes:         make([]*types.BytecodeTrace, 0, len(env.Codes)),
 		StorageTrace:      env.StorageTrace,
 		ExecutionResults:  env.ExecutionResults,
 		TxStorageTraces:   env.TxStorageTraces,
 		Transactions:      txs,
 		StartL1QueueIndex: env.StartL1QueueIndex,
 	}
 
-	for i, tx := range block.Transactions() {
-		evmTrace := env.ExecutionResults[i]
-		// Contract is created.
-		if tx.To() == nil {
-			evmTrace.ByteCode = hexutil.Encode(tx.Data())
-		} else { // contract call be included at this case, specially fallback call's data is empty.
-			evmTrace.ByteCode = hexutil.Encode(statedb.GetCode(*tx.To()))
-			// Get tx.to address's code hash.
-			codeHash := statedb.GetPoseidonCodeHash(*tx.To())
-			evmTrace.PoseidonCodeHash = &codeHash
-		}
+	blockTrace.Bytecodes = append(blockTrace.Bytecodes, &types.BytecodeTrace{
+		CodeSize:         0,
+		KeccakCodeHash:   codehash.EmptyKeccakCodeHash,
+		PoseidonCodeHash: codehash.EmptyPoseidonCodeHash,
+		Code:             hexutil.Encode([]byte{}),
+	})
```

Comment:
why do we need this empty value?

Reply:
trace.bytecodes.is_empty() is used to detech trace format before/after this PR.  https://github.com/scroll-tech/zkevm-circuits/blob/ede39de59ce390ca6d3b1f0568e8661e4868f4fc/eth-types/src/l2_types/trace.rs#L18.    So need to make it not empty always.